lib/ansible/modules/system/xfconf.py

```diff
@@ -152,61 +158,78 @@ def main():
     # Assign module values to dictionary values
     channel = module.params['channel']
     property = module.params['property']
-    value_type = module.params['value_type']
-    if module.params['value'].lower() == "true":
-        value = "true"
-    elif module.params['value'] == "false":
-        value = "false"
+    values = module.params['value']
+    value_types = module.params['value_type']
+    if values is not None and value_types is not None:
+        if len(values) != len(value_types):
+            module.fail_json(msg='Same number of "value" and "value_type" needed')
+        for i in range(len(values)):
+            if values[i].lower() == "true" or values[i].lower() == "false":
+                values[i] = values[i].lower()
+        for value_type in value_types:
+            if value_type not in ['int', 'bool', 'float', 'string']:
+                module.fail_json(msg='value_type %s is not supported'
+                                 % str(value_type))
     else:
-        value = module.params['value']
+        values = value_types = None
+    array = module.params['force_array'] or values is not None and len(values) > 1
```

Comment:
```suggestion
    array = module.params['force_array'] or (values is not None and len(values) > 1)
```
to avoid ambiguities when reading